server/s3_object_metadata.cc

```diff
@@ -225,7 +225,7 @@ S3ObjectMetadata::S3ObjectMetadata(
     mote_kv_writer_factory = std::make_shared<S3MotrKVSWriterFactory>();
   }
   obj_type = S3ObjectMetadataType::simple;
-  primary_obj_size = 0;
+  primary_obj_size = request->get_content_length();
```

Comment:
You can have comment saying
This will be initially object size, in case of object fragment this will be fragment size

Reply:
done.